### DISCUSSION TITLE
Disable Threading

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#1404] [Help]
Hello,

I'm wondering if there's any way to overall disable threading for Minestom or to at least disable it for chunk generation.

I need this as my chunk generation code is written in JS and GraalVM [doesn't allow](https://medium.com/graalvm/multi-threaded-java-javascript-language-interoperability-in-graalvm-2f19c1f9c37b) multi-threaded access to JS so the chunk generation just breaks.

**Comment by KrystilizeNevaDies** [comment]
Heya, you could do this with a simple lock.

Example:

```java
Object jsAccessLock = new Object();
Generator generator = unit -> {
  // Generate blocks outside of js access
  synchronised (jsAccessLock) {
    // Do js access here, it'll never be multithreaded
  }
}
```
